### ISSUE TITLE
Cache Busting does not work due to Supported File Types setting

### ISSUE THREAD

**Issue by antonkorotkov** [udx/wp-stateless#377]
When (1) is not Disabled then (2) appears.

For some reason `Utility::randomize_filename()` function considers option (2) while generating new file name if Cache Busting is enabled.

```php
$body_rewrite_types = ud_get_stateless_media()->get( 'sm.body_rewrite_types' );
if(empty($info['extension']) || strpos($body_rewrite_types, $info['extension']) === false){
          return $filename;
}
```

In other words, Cache Busting only works if the file extension is listed in `Supported File Types` setting which appears only if `File URL Replacement` setting is enabled.

`Supported File Types` setting was set to `false` on the site where this issue first appeared. It was on Network Settings. The `false` value is probably a legacy of this https://github.com/wpCloud/wp-stateless/issues/369#issuecomment-476118354

![image](https://user-images.githubusercontent.com/2276882/56646334-07906600-6688-11e9-8484-7dec1031c649.png)

**Comment by alimuzzaman** [comment]
@ewsopp 
we are ignoring `Cache-Busting` for any file except those in supported file types.

- Should we keep using `Supported File Types` with fallback default if the field is empty?
- Or should we introduce **new field** Exclude file types for `Cache-Busting`?

**Comment by ewsopp** [comment]
The "Cache-Busting" and "File URL Replacement" are separate features. They have no relationship with each other.

**Comment by alimuzzaman** [comment]
@ewsopp then should I remove this code and handle in compatibility?
If removed we have to test with Woocommerce export feature #302.

**Comment by ewsopp** [comment]
Remove what code? 

Again, these features are different and should not affect each other. Read the descriptions of the settings three times, or more, to fully understand.

**Comment by alimuzzaman** [comment]
@ewsopp 
> Remove what code?

At the top in issue description.

That code make sure that `Cache-Busting` feature don't randomize the file names that are not in the `Supported File Types` filed.
I had wrote that for issue #302.

That's why you could export gravity form when you enabled `File URL Replacement`.

**Comment by alimuzzaman** [comment]
Removed the conflicting code and made changes so that fix for issue #302 (woocommerce product export) works.
